tools/c7n_azure/c7n_azure/utils.py

```diff
@@ -417,9 +418,13 @@ def parse_ip_ranges(data, key):
         ranges = [[s.strip() for s in r.split('-')] for r in data[key]]
         result = IPSet()
         for r in ranges:
-            if len(r) > 2:
-                raise Exception('Invalid range. Use x.x.x.x-y.y.y.y or x.x.x.x or x.x.x.x/y.')
-            result.add(IPRange(*r) if len(r) == 2 else IPNetwork(r[0]))
+            resolved_set = resolve_service_tag_alias(r[0])
```

Comment:
Can you add a test for the invalida case?

Reply:
Annnnnd it doesn't work right (if its invalid + you have other IP's in the array) -- good catch :)

Reply:
added a bunch of tests for this and fixed bug